### DISCUSSION TITLE
"500 Server Error" occurred when the data set was permanently deleted.

### DISCUSSION THREAD

**Discussion by hongjie123** [ckan/ckan#7079] [Q&A]
According to this document (https://docs.ckan.org/en/2.5/sysadmin-guide.html#permanently-deleting-datasets), 500 Server Error occurs during the Permanently Delete Dataset operation.
![image](https://user-images.githubusercontent.com/54384385/189589019-1a38acf0-d2f1-4c22-8781-596db213243c.png)

![image](https://user-images.githubusercontent.com/54384385/189589525-4f8501fb-d866-4f77-bf40-386625a790ca.png)

**Comment by hongjie123** [comment]
The log file "ckan_default.error.log" displays the following results:
![image](https://user-images.githubusercontent.com/54384385/189605880-dc4bc241-bc4d-4ee0-8710-af4e7257fb75.png)

I checked my database and there is no "tool_revision" table, but there is a "package_revision" table. I would like to try to modify the SQL statement in the figure that reported an error by asking him to visit "package_revision". However, no SQL statement was found in the program.
![image](https://user-images.githubusercontent.com/54384385/189607378-23b0674e-2e27-4ec1-9211-a613d3cd547b.png)

**Comment by amercader** [comment]
Looks like you are using an old CKAN version that still uses VDM. I think trying to debug this will be really difficult (somehow VDM is getting the dataset type mixed up with the table names). I think it will be easier if you manually update the `package_revision` table to change all `type=tool` rows to `type=dataset` and try to purge again. Do a backup of the database first!
And please consider upgrading to at least CKAN 2.9

**Comment by hongjie123** [comment]
Because the content is different, my site distinguishes between tools and datasets. Therefore, you cannot manually update Type = Tool in the package_Revision table to Type = Dataset. Is there any other solution?
